### PR TITLE
Map Constructor should not allow duplicate keys

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/issues/2203

### CODE REVIEW

File: .idea/libraries/expath.xml [hunk 1]
```diff
@@ -4,7 +4,9 @@
       <root url="file://$PROJECT_DIR$/extensions/expath/lib" />
     </CLASSES>
     <JAVADOC />
-    <SOURCES />
+    <SOURCES>
+      <root url="file://$PROJECT_DIR$/../expath-http-client-java/http-client-java/src/main/java" />
+    </SOURCES>
```
Comment (joewiz) [inline .idea/libraries/expath.xml]: Was this intended to be included?

Comment (adamretter) [reply .idea/libraries/expath.xml]: Oops. Have now fixed.

Comment (joewiz) [reply .idea/libraries/expath.xml]: ;)

File: .idea/libraries/expath.xml [hunk 2]
```diff
@@ -7,4 +7,4 @@
     <SOURCES />
     <jarDirectory url="file://$PROJECT_DIR$/extensions/expath/lib" recursive="false" />
   </library>
-</component>
\ No newline at end of file
+</component>
```
Comment (joewiz) [inline .idea/libraries/expath.xml]: No biggie, but apparently a small whitespace change remained...

**Comment by wolfgangmm** [comment]
Somehow all the travis builds seem to be hanging?

**Comment by adamretter** [comment]
@wolfgangmm It is not hung it is queued. See - https://travis-ci.com/eXist-db/exist/builds

I am not sure if after moving from travis-ci.org to travis-ci.com we have less build workers available, but it looks like that way, all tasks used to build in parallel, but now it looks like just 2 workers. Maybe @duncdrum knows something?

**Comment by duncdrum** [comment]
@adam I noticed that as well. I m not aware of any previous arrangements to get more workers? We are now at what I understand to be the default for oss.

**Pull request merged**